### ISSUE TITLE
Get result call larg data RFC

### ISSUE THREAD

**Issue by lucasmichel** [gkralik/php7-sapnwrfc#110]
First sorry my English :) 

A customer has an RFC that returns a large amount of data.
When testing direct execution in SAP, the RFC returns data.
When I execute using the SAPNWRFC connection, I simply cannot capture the return, nor cause any failure, the call is simply on hold. Observe in the image below the execution time of the RFC in the SAP port, this query returns an average of 3 million and 500 thousand records.

![SAP](https://user-images.githubusercontent.com/3743280/117460361-d3143d80-af22-11eb-8924-b965a3b80927.jpeg)

Is there any way to resolve this issue?

**Comment by gkralik** [comment]
Can you provide more details? At least a trace file would be interesting. How long did you let the script run?

**Comment by lucasmichel** [comment]
@gkralik I have a trace activated with TRACE_LEVEL_FULL, I took a look at it and found no possible errors.
The interesting thing is that when we execute the RFC through SAPGui it takes about 35 minutes to return the data, when I do the query using php with sapnwrfc, let it run for more than 12 hours and not obitive response neither success nor failure , and as if SAP simply did not respond to the request.
I asked the personnel responsible for the sap to also monitor the execution. If you have any feedback from them, I’ll let you know here.

**Comment by lucasmichel** [comment]
@gkralik I did a new test here, the data could get to PHP but now this error has occurred:

![image](https://user-images.githubusercontent.com/3743280/118035940-50381c00-b342-11eb-84b3-ca5e90335673.png)

Is there any alternative for you to prevent this?

**Comment by gkralik** [comment]
Well, it looks like the SDK buffers the data somewhere locally and runs out of space. You'll have to investigate further to see whats happening.

**Comment by lucasmichel** [comment]
Hello, @gkralik  I've tried everything.
I unstet the object and connection, implemented __destruct with 

$this->resCon->close ();
unset ($ this-> resCon);
gc_collect_cycles ();


and yet the memory of the query is not released. I changed the rfc to receive the amount of records that I want it to return, I am bringing 200000 records per query but at the end of the execution the memory is never released causing memory overflow whenever I use the sapnwrf invoke method.
During processing I can control and free memory using gc_collect_cycles ();

But the data that comes from the invoke method is not released, it is always accumulated.
